x-pack/plugins/security_solution/public/management/components/endpoint_execute_action/execute_action_host_response_output.tsx

```diff
@@ -99,42 +96,72 @@ interface ExecuteActionHostResponseOutputProps {
 }
 
 export const ExecuteActionHostResponseOutput = memo<ExecuteActionHostResponseOutputProps>(
-  ({ action, agentId = action.agents[0], 'data-test-subj': dataTestSubj, textSize }) => {
+  ({ action, agentId = action.agents[0], 'data-test-subj': dataTestSubj, textSize = 'xs' }) => {
+    const { canWriteExecuteOperations } = useUserPrivileges().endpointPrivileges;
+
+    const [executeOutputContent, setExecuteOutputContent] = useState<
+      undefined | ResponseActionExecuteOutputContent
+    >(undefined);
+
     const outputContent = useMemo(
       () =>
         action.outputs &&
         action.outputs[agentId] &&
         (action.outputs[agentId].content as ResponseActionExecuteOutputContent),
-      [agentId, action]
+      [action.outputs, agentId]
     );
 
+    const { data: actionDetails, isFetching } = useGetActionDetails(action.id, {
+      enabled: canWriteExecuteOperations && !outputContent,
+    });
+
+    useEffect(() => {
+      if (
+        !isFetching &&
+        actionDetails &&
+        actionDetails.data &&
+        actionDetails.data.outputs &&
+        actionDetails.data.outputs[agentId]
+      ) {
+        setExecuteOutputContent(
+          actionDetails.data.outputs[agentId].content as ResponseActionExecuteOutputContent
+        );
+      }
+
+      return () => {
+        setExecuteOutputContent(undefined);
+      };
+    }, [actionDetails, agentId, isFetching]);
+
+    if (!canWriteExecuteOperations || !executeOutputContent) {
```

Comment:
Is this correct? If the component is also being used in the Action History Log, then this might be a problem because a user might not actually have `canWriteExecuteOperations` but have access to the History Log.

Reply:
I changed this now to only use `!executeOutputContent ` condition